frontend/packages/shared/src/contexts/ServicesContext.test.tsx

```diff
@@ -55,12 +53,29 @@ describe('ServicesContext', () => {
       },
     );
     expect(await screen.findByText(textMock('api_errors.Unauthorized'))).toBeInTheDocument();
-    jest.runAllTimers();
-    await waitFor(() => {
-      expect(queriesMock.logout).toHaveBeenCalledTimes(1);
-    });
+  });
 
-    expect(mockConsoleError).toHaveBeenCalled();
+  it('logs out the user and redirects after the unauthorized error toast disappears', async () => {
+    const logoutMock = jest.fn().mockResolvedValueOnce(Promise.resolve());
+
+    const { result } = renderHook(
+      () =>
+        useQuery({
+          queryKey: ['fetchData'],
+          queryFn: () => Promise.reject(createApiErrorMock(401)),
+          retry: false,
+        }),
+      {
+        wrapper: ({ children }) => wrapper({ children, queries: { logout: logoutMock } }),
+      },
+    );
+
+    await waitFor(() => result.current.isError);
+
+    await waitFor(() => screen.findByText(textMock('api_errors.Unauthorized')));
+
+    jest.runAllTimers();
+    // TODO check if the user is logged out.
```

Comment:
@mlqn I tried to call logout by ` expect(logoutMock).toHaveBeenCalled();` but did not worked , any tips?

Reply:
Hmm... not sure 🤔 I'll look into it

Reply:
I think no need , It works now 🙂🤞

Reply:
https://github.com/fkhadra/react-toastify/issues/1010

Reply:
After a quick discussion with @mlqn, we might revert to handling the timeout ourselves. Then we can write tests for this use-case.